cardshifter-server/src/test/java/com/cardshifter/server/main/ServerConnectionTest.java

```diff
@@ -252,5 +252,16 @@ public void testUsersQueryNotLoggedIn() throws IOException, InterruptedException
 		// There shouldn't be a UserStatusMessage for client2
 		assertNoMessage(client1);
 	}
+
+	@Test(timeout = 5000)
+	public void testNoOfflineMessageIfNotLoggedIn() throws IOException, InterruptedException {
+		System.out.println("testNoOfflineMessageIfNotLoggedIn");
+		TestClient client2 = createTestClient();
+		client2.disconnect();
+
+		// Avoid race condition between UserStatusMessage and ServerStatusMessage sent by assertNoMessage
+		Thread.sleep(1000);
```

Comment:
When I wrote the test it was supposed to fail because I hadn't fixed the bug yet. But when I ran the test suite it was sometimes passing and sometimes not. 

This unpredictability was because of the order of received messages: `client2.disconnect()` triggered a `UserStatusMessage`, while `assertNoMessage` triggered a `ServerStatusMessage`. `assertNoMessage` was added in #392, discussed [here](http://chat.stackexchange.com/rooms/16134/conversation/assert-no-message), and relies on the order of messages.

If you want to see the race yourself, check out the first commit of this PR f94887a751dadbf770d76e311e862b070352dc92 and remove this line. (You might or might not see this problem on your specific machine)